recipes/ncbi-cxx-toolkit-public/all/conanfile.py

```diff
@@ -0,0 +1,293 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration, ConanException
+from conan.tools.microsoft import check_min_vs, is_msvc_static_runtime, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
+from conan.tools.build import check_min_cppstd, cross_building
+from conan.tools.scm import Version
+from conan.tools.cmake import CMakeDeps, CMakeToolchain, CMake, cmake_layout
+import os
+import yaml
+
+required_conan_version = ">=1.53.0"
+
+
+class NcbiCxxToolkit(ConanFile):
+    name = "ncbi-cxx-toolkit-public"
+    description = "NCBI C++ Toolkit -- a cross-platform application framework and a collection of libraries for working with biological data."
+    license = "CC0-1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://ncbi.github.io/cxx-toolkit"
+    topics = ("ncbi", "biotechnology", "bioinformatics", "genbank", "gene",
+              "genome", "genetic", "sequence", "alignment", "blast",
+              "biological", "toolkit", "c++")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared":     [True, False],
+        "fPIC":       [True, False],
+        "with_projects": ["ANY"],
+        "with_targets":  ["ANY"],
+        "with_components": ["ANY"]
+    }
+    default_options = {
+        "shared":     False,
+        "fPIC":       True,
+        "with_projects":  "",
+        "with_targets":   "",
+        "with_components": ""
+    }
+    short_paths = True
+    tk_dependencies = None
+    tk_requirements = None
+    tk_componenttargets = set()
```

Comment:
```suggestion
    _tk_dependencies = None
    _tk_requirements = None
    _tk_componenttargets = set()
```

Custom user variables in recipes need to start with underscore, see https://docs.conan.io/2/knowledge/guidelines.html?highlight=public#forbidden-practices 

This is why the recipe is failing. (Conan expects to be able to serialize all "public" members, and the `set()` was breaking it)